notebooks/concepts/Specviz from splot concept notebook.ipynb

```diff
@@ -0,0 +1,3636 @@
+    "\n",
+    "1. For a notebook, most of the fitting-related functionality is best implemented via specutils and Astropy's model-fitting machinery applied to the specviz spectrum. All but the specviz-accessing parts are [ALREADY IMPLEMENTED]. E.g., to fit a order-12 legendre polynomial to a continuum while excluding the H$\\alpha$ region, and then using it to subtract the continuum from the existing specviz spectrum, one could do:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "continuum = specutils.fitting.fit_continuum(specviz.current_spectrum, \n",
+    "                                            model=astropy.modeling.models.Legendre1D(12),\n",
+    "                                            exclude_regions=[specutils.SpectralRegion.from_center(6563*u.angstrom, 30*u.angstrom)])\n",
+    "\n",
+    "continuum_subtracted_spectrum = specviz.current_spectrum - continuum(specviz.current_spectrum.spectral_axis)\n",
+    "\n",
+    "specviz.load_data(continuum_subtracted_spectrum)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    An almost-identical interface can be used to fit spectral lines.  For example, to fit a voigt profile to the most-recently selected subset:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "specutils.fitting.fit_lines(specviz.current_spectrum,\n",
+    "                            astropy.modeling.models.Voigt1D(),\n",
+    "                            window=specviz.get_selected_region())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. for the desktop interface, A fitting UI is necessary for the above steps, which is currently in-progress for cubeviz.  The same model-fitting UI should be adopted for specviz.  Missing pieces that are needed to replicate SPLOT functionality include:\n",
+    "\n",
+    "    * A way to generate a new spectrum in specviz that comes from subtracting a fitted continuum from the current spectrum (overlaps with the arithmetic discussion below)\n",
+    "    * A way to generate a new spectrum in specviz that comes from dividing by a fitted continuum from the current spectrum (overlaps with the arithmetic discussion below)\n",
+    "    * Functionality to allow sigma-clipping as part of the fitting process (bonus/stretch goal to actually *show* the clipped points)\n",
+    "    * Functionality to allow manual rejection of specific points to be excluded in the fit.\n",
+    "    \n",
+    "    Note that the first two are available trivially from the notebook interface.  The latter two area also possible from a notebook but not necessarily easy."
+   ]
+  },
```

Comment:
Don't know if it is in SPLOT, but it would be nice fit a cubic spline after specifying multiple knot points and exclusion (or fitting) windows.

Reply:
This is definitely possible in the notebook.  I'm less convinced this is an important feature in the UI, so I'm going to leave it off but maybe you can make a separate issue specifically to discuss this @PatrickOgle ?